src/code/leafletClasses.js

```diff
@@ -37,17 +37,18 @@ export const WTooltip = L.Class.extend({
 });
 
 export const WDialog = L.Handler.extend({
-  initialize: function (map = window.map, options) {
-    if (this.type === undefined) this.type = "Unextended Wasabee Dialog";
-    this._map = map;
-    this._container = map._container;
-    this.options = {};
-    L.Util.extend(this.options, options);
+  statics: {
+    TYPE: "Unextended Wasabee Dialog",
+  },
+
+  initialize: function (options) {
+    L.setOptions(this, options);
     this._enabled = false;
     this._smallScreen = this._isMobile();
     this._dialog = null;
     // look for operation in options, if not set, get it
     // determine large or small screen dialog sizes
+    window.map.fire("wdialog", this);
   },
```

Comment:
Would it be helpful to register on window.map.on("wasabeUIUpdate", this._update()) in addHooks

and define a default
_update: function () { };

and unregister it on removeHooks ..

For the dialogs that need _update support, they simply override the empty _update? Or is that unnecessary?